tests/plugins/test_gen_json.py

```diff
@@ -0,0 +1,150 @@
+
+from unittest import TestCase
+from tempfile import TemporaryDirectory, NamedTemporaryFile
+
+from aac.plugins.gen_json.gen_json_impl import print_json
+from aac.plugins.plugin_execution import PluginExecutionStatusCode, plugin_result
+
+
+class Testgen_json(TestCase):
+
+    def test_print_json_with_output_directory(self):
+
+        with TemporaryDirectory() as temp_dir, NamedTemporaryFile("w") as temp_arch_file:
+            temp_arch_file.write(TEST_ARCH_YAML_STRING)
+            temp_arch_file.seek(0)
+
+            with plugin_result("", print_json, [temp_arch_file.name], temp_dir) as result:
+                self.assertEqual(result.status_code, PluginExecutionStatusCode.SUCCESS)
+
+    def test_gen_json_output_to_cli(self):
+
+        with TemporaryDirectory(), NamedTemporaryFile("w") as temp_arch_file:
+            temp_arch_file.write(TEST_ARCH_YAML_STRING)
+            temp_arch_file.seek(0)
+
+            with plugin_result("", print_json, [temp_arch_file.name]) as result:
+                self.assertEqual(result.status_code, PluginExecutionStatusCode.SUCCESS)
+
+    def test_gen_json_with_invalid_arch_file(self):
+        pass
+
+
+TEST_ARCH_YAML_STRING = """
```

Comment:
Yaml String shortened to one model and three data types. The original was copied from test_gen_protobuf.py. We will want to remember to shorten it there as well.

Reply:
shortened the YAML string. The original string was copied from test_gen_protobuf.py, might want to think about shortening it there as well.

Reply:
The `test_gen_protobuf.py` YAML string may be needed for that because that's actually doing stuff with the different YAML "declarations" but here, we just want to see that the YAML gets converted to JSON, it doesn't matter much what the YAML actually is (as long as it validates).